url4aac7e6f55_simon0191_tetris3d_tgz-pJ8-Tetris3dJ8/src/vsdk/toolkit/environment/geometry/Box.java

```diff
@@ -0,0 +1,270 @@
+//===========================================================================
+//=-------------------------------------------------------------------------=
+//= Module history:                                                         =
+//= - February 12 2006 - Oscar Chavarro: Original base version              =
+//=-------------------------------------------------------------------------=
+//= References:                                                             =
+//= [MANT1988] Mantyla Martti. "An Introduction To Solid Modeling",         =
+//=     Computer Science Press, 1988.                                       =
+//===========================================================================
+
+package vsdk.toolkit.environment.geometry;
+
+import vsdk.toolkit.common.VSDK;
+import vsdk.toolkit.common.linealAlgebra.Vector3D;
+import vsdk.toolkit.environment.geometry.Geometry;
+import vsdk.toolkit.environment.geometry.GeometryIntersectionInformation;
+import vsdk.toolkit.common.Ray;
+
+public class Box extends Solid {
+    /// Check the general attribute description in superclass Entity.
+    public static final long serialVersionUID = 20060502L;
+
+    private Vector3D size;
+
+    private GeometryIntersectionInformation lastInfo;
+    private int lastPlane;
+
+    public Box(double dx, double dy, double dz) {
+        size = new Vector3D(dx, dy, dz);
+
+        lastInfo = new GeometryIntersectionInformation();
+        lastPlane = 0;
+    }
+
+    public Box(Vector3D s) {
+        size = new Vector3D(s);
+
+        lastInfo = new GeometryIntersectionInformation();
+        lastPlane = 0;
+    }
+
+    public int getLastIntersectedPlane()
+    {
+        return lastPlane;
+    }
+
+    /**
+     Check the general interface contract in superclass method
+     Geometry.doIntersection.
+    */
+    public boolean
+    doIntersection(Ray inOutRay) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 43. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 109-157 into a separate method.